### ISSUE TITLE
[FE][Bug] Contact Info: Edit mode is not exited when moving between Contact Info and other Profile sections

### ISSUE THREAD

**Issue by MarciMcGuireGCIO** [department-of-veterans-affairs/va.gov-team#32114]
1. Sign into staging with user 378
2. Go to https://staging.va.gov/profile/personal-information
3. Click the Edit button in the Mailing address section
4. Navigate to another page _within_ the Profile (Military info, DIrect deposit info, etc.)
5. Return to the Personal and contact information section
6. Notice that we are still in "edit" mode

Note - this video demonstrates the issue when moving from the Profile Direct deposit section to another, but the same issue applies on the Personal Information page https://user-images.githubusercontent.com/20728956/121097762-62727200-c7a9-11eb-81a4-bf05888a9c69.mp4

This does not happen when you navigate to a page _outside_ of the Profile and come back since that triggers a full page refresh.

## Expected behavior
No fields should be in edit mode when you first load any section of the Profile

## Possible fix
Exit edit mode when the component in question "unmounts." We could likely accomplish this using the "cleanup" function returned by a `useEffect` hook.

**Comment by ajakabcin** [comment]
Closely related to #25788

**Comment by hdjustice** [comment]
@tmitchellgcio I do not really understand Pull Requests (might need a crash course!), but looks like this one got approved.  Can you check on it this morning and let us know if we need to get some other folks on it?  Thanks!
cc @ajakabcin

**Comment by tmitchellgcio** [comment]
@hdjustice PRs are just changes to a branch off of the main codebase. Before they can be merged into the main branch they have to be approved to make sure we didn't do anything that breaks something else. I have one small change to that PR and then I can merge it

**Comment by hdjustice** [comment]
Thank you @tmitchellgcio for the explanation, that helps alot!  Let me know if you need any support.

**Comment by hdjustice** [comment]
Validated in staging environment that the issue has been corrected.  When I selected to edit a section within Personal and Contact information, navigated to another section of Profile (used both Military Information, and Direct Deposit information tabs), and navigated back to Personal and Contact information, the sections I selected to edit were back in their original read-only state, no longer in edit state.  This is expected behavior.  Ticket will be closed.
